internal/process/provisioning/resolve_creds_hap_selection_migration_test.go

```diff
@@ -0,0 +1,226 @@
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/kyma-project/kyma-environment-broker/common/hyperscaler"
+	"github.com/kyma-project/kyma-environment-broker/common/hyperscaler/rules"
+	"github.com/kyma-project/kyma-environment-broker/internal/storage"
+	"github.com/stretchr/testify/assert"
+
+	k8sTesting "k8s.io/client-go/testing"
+)
+
+// To generate output hap-old-implementation files (used to compare results by hand) switch writeFiles to true
+func TestResolveCredentials_IntegrationAzure2(t *testing.T) {
+	writeFiles := false
+
+	memoryStorage := storage.NewMemoryStorage()
+
+	unstructuredList := &unstructured.UnstructuredList{
+		Items: []unstructured.Unstructured{},
+	}
+
+	var oldImplementationLog *bufio.Writer
+	var newImplementationLog *bufio.Writer
+
+	if writeFiles {
+		oldLog, err := os.OpenFile("./hap-old-implementation.log", os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0755)
+		require.NoError(t, err)
+		defer oldLog.Close()
+
+		oldImplementationLog = bufio.NewWriter(oldLog)
+
+		newLog, err := os.OpenFile("./hap-new-implementation.log", os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0755)
+		require.NoError(t, err)
+		defer newLog.Close()
+
+		newImplementationLog = bufio.NewWriter(newLog)
+	}
+
+	gc := gardener.NewDynamicFakeClient()
+	gc.PrependReactor("get", gardener.SecretBindingResource.Resource, func(action k8sTesting.Action) (bool, runtime.Object, error) {
+		return true, fixDummySecretBinding(""), nil
+	})
+
+	// in memory storage for queried labels and results, used for
+	// asserting if the same labels do not produce different secrets queries
+	labelsResultsMap := map[string]*unstructured.Unstructured{}
+
+	// used to compare labels between two runs
+	// key - string - is originally queried label
+	// value - int - is used to check how many times a secret has been queried
+	queriesCountMap := map[string]int{}
+	increase := true
+
+	gc.PrependReactor("list", gardener.SecretBindingResource.Resource, func(action k8sTesting.Action) (bool, runtime.Object, error) {
+		originalLabel := action.(k8sTesting.ListActionImpl).GetListRestrictions().Labels.String()
+
+		if increase {
+			if writeFiles {
+				n, err := oldImplementationLog.WriteString(originalLabel + "\n")
+				require.NoError(t, err)
+				require.NotZero(t, n)
+			}
+		} else {
+			if writeFiles {
+				n, err := newImplementationLog.WriteString(originalLabel + "\n")
+				require.NoError(t, err)
+				require.NotZero(t, n)
+			}
+		}
+
+		// count query times
+		if _, ok := queriesCountMap[originalLabel]; !ok {
+			queriesCountMap[originalLabel] = 1
+		} else {
+			if increase {
+				queriesCountMap[originalLabel]++
+			} else {
+				queriesCountMap[originalLabel]--
+			}
+		}
+
+		if _, ok := labelsResultsMap[originalLabel]; !ok {
+
+			// filter out labels that select secrets without that label
+			// for example !dirty selects a secret without dirty label
+			filteredSplit := []string{}
+			split := strings.Split(originalLabel, ",")
+			for _, label := range split {
```

Comment:
This part is needed only to fix appropriate SecretBinding.
So could be extracted as it has different responsibility.